File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/annotations/OpenApiContentInstance.java
package net.dzikoysk.openapi.processor.annotations;

import io.javalin.plugin.openapi.annotations.OpenApiContent;
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper;

import javax.lang.model.element.AnnotationMirror;
import javax.lang.model.type.TypeMirror;
import java.lang.annotation.Annotation;

public final class OpenApiContentInstance extends AnnotationMirrorMapper {

    public OpenApiContentInstance(AnnotationMirror mirror) {
        super(mirror);
    }

    public TypeMirror from() {
        return getType("from");
    }

    public boolean isArray() {
        return getBoolean("isArray");
    }

    public String type() {
        return getString("type");
    }

    public Class<? extends Annotation> annotationType() {
        return OpenApiContent.class;
    }

}

File: javalin-openapi-processor/src/main/kotlin/net/dzikoysk/openapi/processor/annotations/OpenApiContentInstance.kt
package net.dzikoysk.openapi.processor.annotations

import io.javalin.plugin.openapi.annotations.OpenApiContent
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper
import javax.lang.model.element.AnnotationMirror
import javax.lang.model.type.TypeMirror

class OpenApiContentInstance(mirror: AnnotationMirror) : AnnotationMirrorMapper(mirror) {

    fun from(): TypeMirror =
        getType("from")

    fun isArray(): Boolean =
        getBoolean("isArray")

    fun type(): String =
        getString("type")

    fun annotationType(): Class<out Annotation> =
        OpenApiContent::class.java

}